rotkehlchen/externalapis/alchemy.py

```diff
@@ -0,0 +1,421 @@
+            elif asset.chain_id == ChainID.SCROLL:
+                network = 'scroll-mainnet'
+            elif asset.chain_id == ChainID.GNOSIS:
+                network = 'gnosis-mainnet'
+            elif asset.chain_id == ChainID.BASE:
+                network = 'base-mainnet'
+            elif asset.chain_id == ChainID.FANTOM:
+                network = 'fantom-mainnet'
+            elif asset.chain_id == ChainID.ARBITRUM_NOVA:
+                network = 'arbnova-mainnet'
+            else:
+                raise UnsupportedAsset(asset.identifier)
+
+            return AlchemyAssetData(
+                network=network,
+                address=asset.evm_address,
+            )
+
+        return AlchemyAssetData(symbol=asset.symbol)
+
+    @staticmethod
+    def _deserialize_current_price(
+            result: list[dict[str, Any]],
+            alchemy_asset_data: AlchemyAssetData,
+            from_asset: Asset,
+            to_asset: Asset,
+    ) -> Price:
+        """Deserializes current price data from Alchemy API response.
+
+        If price data is missing, invalid, or there's an error, ZERO_PRICE is returned.
+        Otherwise, the deserialized price is returned.
+        """
+        usd_price = ZERO_PRICE
+        try:
+            for entry in result:
+                if not alchemy_asset_data.matches_api_response(entry):
+                    continue
+
+                if 'error' in entry:
+                    log.warning(
+                        f'Queried Alchemy Price API from {from_asset.identifier} to {to_asset.identifier}. '  # noqa: E501
+                        f'API returned error: {entry["error"]}',
+                    )
+                    return usd_price
+
+                if len(prices := entry['prices']) == 0:
+                    log.warning(
+                        f'Queried Alchemy Price API from {from_asset.identifier} to {to_asset.identifier}, '  # noqa: E501
+                        f'but received no prices in response',
+                    )
+                    return usd_price
+
+                for price in prices:
+                    if price['currency'] == 'usd':
+                        usd_price = deserialize_price(price['value'])
+                        break
+                else:  # usd price not found
+                    log.error(
+                        f'Queried Alchemy Price API from {from_asset.identifier} to '
+                        f'{to_asset.identifier}, but received non-USD price response',
+                    )
+        except (DeserializationError, KeyError) as e:
+            error_msg = f'missing key {e!s}' if isinstance(e, KeyError) else str(e)
+            log.warning(
+                f'Queried Alchemy Price API from {from_asset.identifier} to {to_asset.identifier}. '  # noqa: E501
+                f'Failed to deserialize price data due to: {error_msg}',
+            )
+            return usd_price
+        else:
+            return usd_price
```

Comment:
Don't you return usd_price always? Which is equivalent to this?

But isn't there a bug here?

```suggestion
        finally:
            return usd_price
```